### PR TITLE
wxMSW: Toolbar checked item bg color fix for dark mode

### PR DESCRIPTION
**Pull request by sz5000**
In dark mode and MSW, a checked toolbar item always has this light blue background color as you can see here:

image

This pull request fixes this and the new color better fits the dark design, as you can see here:

image

The flag `CDRF_NOTIFYPOSTPAINT` is actually not required for this change, but but you accept another pull request (#24901) that fixes the dark dropdown arrow issue in dark theme, then it is required.

### CODE REVIEW

**Comment by vadz** [comment]
Thanks for the fix, but do you understand why the problem isn't visible in the toolbar sample? It has a checkable icon (weirdly enough, it's the "help" one) and it appears correctly in the dark mode. What is different about the item in your case?

Also, if we do need this fix, shouldn't we restrict it to the dark mode only?

**Comment by sz5000** [comment]
Seems the be different on Windows 11 and 10.

Inside a Windows 10 Test VM:

image

Under Windows 11:

image

Maybe it would be a good decision to make it customizable to that the dev can change this color.


Regarding "shouldn't we restrict it to the dark mode only":

Yes, it already is, 

```
case CDDS_PREPAINT:
    if ( !wxMSWDarkMode::IsActive() )
        break;

    *result = CDRF_NOTIFYITEMDRAW;
    return true;
```

CDRF_NOTIFYITEMDRAW is only set, if dark mode is active, so `case CDDS_ITEMPREPAINT` is only executed in dark mode.

**Comment by sz5000** [comment]
I made another change and I think it's ok now and look good on Windows 10 and 11.  Please have a look.

File: src/msw/toolbar.cpp [hunk 1]
```diff
@@ -1694,6 +1694,16 @@ bool wxToolBar::MSWOnNotify(int WXUNUSED(idCtrl),
                 nmtbcd->clrTextHighlight = wxColourToRGB(GetForegroundColour());
                 nmtbcd->clrHighlightHotTrack = wxSysColourToRGB(wxSYS_COLOUR_HOTLIGHT);
 
+                // draw custom checked button background
+                if ((nmtbcd->nmcd.uItemState & (CDIS_CHECKED|CDIS_HOT)) == CDIS_CHECKED) {
```
Comment (PBfordev) [inline src/msw/toolbar.cpp:1698]:
```suggestion
                if ( (nmtbcd->nmcd.uItemState & (CDIS_CHECKED | CDIS_HOT)) == CDIS_CHECKED ) 
                {
```

File: src/msw/toolbar.cpp [hunk 2]
```diff
@@ -1694,6 +1694,16 @@ bool wxToolBar::MSWOnNotify(int WXUNUSED(idCtrl),
...
+                    wxColor color = wxSystemSettings::GetColour(wxSYS_COLOUR_HOTLIGHT).ChangeLightness(110);
```
Comment (PBfordev) [inline src/msw/toolbar.cpp:1699]:
```suggestion
                    const wxColor color = wxSystemSettings::GetColour(wxSYS_COLOUR_HOTLIGHT).ChangeLightness(110);
```

File: src/msw/toolbar.cpp [hunk 3]
```diff
@@ -1694,6 +1694,16 @@ bool wxToolBar::MSWOnNotify(int WXUNUSED(idCtrl),
...
+                    HBRUSH br = CreateSolidBrush(wxColourToRGB(color));
+                    FillRect(nmtbcd->nmcd.hdc, &nmtbcd->nmcd.rc, br);
```
Comment (PBfordev) [inline src/msw/toolbar.cpp:1701]:
```suggestion
                    ::FillRect(nmtbcd->nmcd.hdc, &nmtbcd->nmcd.rc, br);
```

File: src/msw/toolbar.cpp [hunk 4]
```diff
@@ -1694,6 +1694,16 @@ bool wxToolBar::MSWOnNotify(int WXUNUSED(idCtrl),
                 nmtbcd->clrTextHighlight = wxColourToRGB(GetForegroundColour());
                 nmtbcd->clrHighlightHotTrack = wxSysColourToRGB(wxSYS_COLOUR_HOTLIGHT);
 
+                // draw custom checked button background
+                if ((nmtbcd->nmcd.uItemState & (CDIS_CHECKED|CDIS_HOT)) == CDIS_CHECKED) {
+                    wxColor color = wxSystemSettings::GetColour(wxSYS_COLOUR_HOTLIGHT).ChangeLightness(110);
+                    HBRUSH br = CreateSolidBrush(wxColourToRGB(color));
```
Comment (PBfordev) [inline src/msw/toolbar.cpp:1700]:
```suggestion
                   AutoHBRUSH br(wxColourToRGB(color));
```

File: src/msw/toolbar.cpp [hunk 5]
```diff
@@ -1694,6 +1694,16 @@ bool wxToolBar::MSWOnNotify(int WXUNUSED(idCtrl),
...
+                    FillRect(nmtbcd->nmcd.hdc, &nmtbcd->nmcd.rc, br);
+                    DeleteObject(br);
```
Comment (PBfordev) [inline src/msw/toolbar.cpp:1702]:
```suggestion
```

**Review by PBfordev — COMMENTED** [review]
Sorry for all the nitpicking (I won't do the same for your other PR), but I believe that if possible, new code should meet the codebase style requirements, in particular, when being inserted into existing code which does.

File: src/msw/toolbar.cpp (hunk 1 above)
Comment (sz5000) [reply src/msw/toolbar.cpp]: I've changed it.

File: src/msw/toolbar.cpp (hunk 2 above)
Comment (sz5000) [reply src/msw/toolbar.cpp]: I've changed it.

File: src/msw/toolbar.cpp (hunk 4 above)
Comment (sz5000) [reply src/msw/toolbar.cpp]: I've changed it.

File: src/msw/toolbar.cpp (hunk 5 above)
Comment (sz5000) [reply src/msw/toolbar.cpp]: I've changed it.

File: src/msw/toolbar.cpp (hunk 3 above)
Comment (sz5000) [reply src/msw/toolbar.cpp]: I've changed it.

**Comment by vadz** [comment]
Thanks! I'm merging it with just an added comment and another minor style change to avoid overlong lines.

P.S. Sorry for missing the existing dark mode check and thanks for pointing it out.

**Comment by sz5000** [comment]
Thanks, looks good so far.

**Pull request closed without merging**